### DISCUSSION TITLE
Specify variable initial condition of new activated element in ActivateElementByPath

### DISCUSSION THREAD

**Discussion by ZoeyChen1993** [idaholab/moose#25863] [Q&A General]
Hello, I am currently working on ActivateElementByPath related work, I am wondering that whether the variable for new activate element  could be specified in the userobject, for example, a pure thermal problem, the element id is transferred from 0 to 1, then could the temperature be also transferred to a specified value as initial condition of this element in the UserObject block instead of defining it in the ics? If I would like to implement this, is there any existing code or what kind of code should I modify or add to the source file?

**Comment by GiudGiud** [comment]
@dewenyushu

**Reply by GiudGiud** [reply]
in my opinion this would not be too hard to do. You just need to modify the user object to set the value of the variable after creating the element. 

or more likely a second user object that acts after the system has been re-sized and sets the value of the variable based on some other variable that distinguishes a new element from an old one (like some sort of boolean field)
